pkg/apis/channels/v1alpha1/subscription_types.go

```diff
@@ -57,14 +63,17 @@ type SubscriptionSpec struct {
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
 
-	// Channel is the name of the channel to subscribe to.
-	Channel string `json:"channel"`
+	// From is the name of the channel to subscribe to for receiving events
+	// to be transformed.
+	From string `json:"from"`
 
-	// Subscriber is the name of the subscriber service DNS name.
-	Subscriber string `json:"subscriber"`
+	// Processor is the processor service DNS name. Events
+	// from the From channel will be delivered here and replies are sent
+	// to To channel.
+	Processor string `json:"processor,omitempty"`
```

Comment:
How about `Call`?

Reply:
since kubernetes resource yaml describes the desired _declarative_ state to be reconciled by a controller, I personally prefer to avoid using **verbs** that suggest the yaml is defining the _imperative_ logic

Reply:
`calls` ?

Reply:
From conversation in Slack, I'd like to see names which provide the following:
1. Clearly unambiguous about the sequence (`from`, `processor`, `to` seems ambiguous whether `to` or `processor` is last).
2. Work cleanly for all three cases (`from`, `processor`, `to`), (`from`, `processor`), (`from`, `to`).

Mostly, we seem to think `from` is clear. Here are some options for the other two:
* `processor`: `call`, `invoke`, `through`, `subscriber`, `apply`
* `to`: `finally` (collides with exception handling), `then` (may not be strong enough), `result`, `out`, `continue` (may collide with loops and not imply last item).

Reply:
Here's an option where there's always a to/from pair at top-level, if both `processor` and `channel` are included in the `to`, the `processor` will always be invoked prior to sending its result to the channel.

```
from: x
to:
  processor: foo
  channel: y

from: x
to:
  processor: foo

from: x
to:
  channel: y
```

The downside is that it adds a level of nesting in the config, but that might be worth the tradeoff since these resources would typically be generated from higher level resources and/or CLIs.

Reply:
What's the upside of adding this extra level of nesting, since to: block will still have the same behaviour as if we kept it the top level?